internal/global/state.go

```diff
@@ -50,14 +51,16 @@ func SetTracerProvider(tp trace.TracerProvider) {
 	delegateTraceOnce.Do(func() {
 		current := TracerProvider()
 		if current == tp {
-			// Setting the provider to the prior default is nonsense, panic.
-			// Panic is acceptable because we are likely still early in the
-			// process lifetime.
-			panic("invalid TracerProvider, the global instance cannot be reinstalled")
+			// Setting the provider to the prior default results in a noop. Return
+			// early.
+			Error(
+				errors.New("no delegate configured in tracer provider"),
+				"Setting tracer provider to it's current value. No delegate will be configured",
+			)
+			return
```

Comment:
When I did this in the global.MeterProvider I pushed this change outside of the sync.Once.  This avoids the problem of preventing the delegate from getting a new provider e.g.:

```
SetTracerProvider(TracerProvider())
SetTracerProvider(sdktrace.NewTracerProvider())
```